### PR TITLE
Create stanza after repohost is added

### PR DESCRIPTION
**Pull request by tony-landreth**
Given a cloud host is already in place,
the user should still be able to add a
repohost.

### CODE REVIEW

File: internal/pgbackrest/pgbackrest_test.go [hunk 1]
```diff
@@ -34,15 +39,19 @@ func TestStanzaCreateOrUpgrade(t *testing.T) {
 	ctx := context.Background()
 	configHash := "7f5d4d5bdc"
 	expectedCommand := []string{"bash", "-ceu", "--", `
-declare -r hash="$1" stanza="$2" message="$3" cmd="$4"
+declare -r hash="$1" stanza="$2" message="$3" cmd="$4" check_repo_cmd="$5"
 if [[ "$(< /etc/pgbackrest/conf.d/config-hash)" != "${hash}" ]]; then
     printf >&2 "%s" "${message}"; exit 1;
+elif ! bash -c "${check_repo_cmd}"; then
```
Comment (tjmoore4) [inline internal/pgbackrest/pgbackrest_test.go:45]: :pick: Could we add a comment to explain why we are checking all the repo paths are in place?

**Review by tjmoore4 — APPROVED** [review]

File: internal/pgbackrest/pgbackrest.go [hunk 2]
```diff
@@ -60,22 +62,44 @@ func (exec Executor) StanzaCreateOrUpgrade(ctx context.Context, configHash strin
...
 	// propagated to the container, and if not, it prints an error and returns with exit code 1).
 	// Otherwise, it runs the pgbackrest command, which will either be "stanza-create" or
 	// "stanza-upgrade", depending on the value of the boolean "upgrade" parameter.
 	const script = `
-declare -r hash="$1" stanza="$2" message="$3" cmd="$4"
+declare -r hash="$1" stanza="$2" message="$3" cmd="$4" check_repo_cmd="$5"
 if [[ "$(< /etc/pgbackrest/conf.d/config-hash)" != "${hash}" ]]; then
     printf >&2 "%s" "${message}"; exit 1;
+elif ! bash -c "${check_repo_cmd}"; then
+		printf >&2 "%s" "${message}"; exit 1;
```
Comment (andrewlecuyer) [inline internal/pgbackrest/pgbackrest.go:95]: ⛏️ Looks like an extra tab in here.

Comment (andrewlecuyer) [inline internal/pgbackrest/pgbackrest.go:95]: 🤔 Should we consider another/different error message here?  The current message specifically specifically references a "config hash mismatch", which really isn't applicable this new check.

At the very least, prob worth adding adding a note/comment in there around the reasoning for leveraging that same error message, e.g. to sure we're not confused with other hashing logic in the future.

Comment (tony-landreth) [reply internal/pgbackrest/pgbackrest.go:95]: Good idea 👍

File: internal/pgbackrest/pgbackrest_test.go (hunk 1 above)
Comment (tony-landreth) [reply internal/pgbackrest/pgbackrest_test.go:45]: Will do.

**Comment by ValClarkson** [comment]
I've tested with multiple repo set ups on creation and it works as expected.  I've hit an error when i remove a repo thats an s3 bucket then you run backrest info this error occurs "error (missing stanza path)" for that repo that was removed.

File: internal/pgbackrest/pgbackrest.go [hunk 3]
```diff
@@ -60,22 +62,44 @@ func (exec Executor) StanzaCreateOrUpgrade(ctx context.Context, configHash strin
...
 	}
 
+	var reposWithVolumes []v1beta1.PGBackRestRepo
+	for _, repo := range postgresCluster.Spec.Backups.PGBackRest.Repos {
+		if repo.Volume != nil {
+			reposWithVolumes = append(reposWithVolumes, repo)
+		}
+	}
+
+	grep := "grep %s-path /etc/pgbackrest/conf.d/pgbackrest_instance.conf"
```
Comment (tony-landreth) [inline internal/pgbackrest/pgbackrest.go:72]: To the point @ValClarkson made, this will handle repos added, but not repos removed.

Comment (tony-landreth) [inline internal/pgbackrest/pgbackrest.go:72]: @andrewlecuyer If we checked that the number of repos in the container config doesn't equal what's in the spec, wouldn't we get the added benefit of handling cases where a repo is removed (regardless of the type of repo)? The hash would still cover the cases where number of repos stays the same, but endpoint changes, right?

File: internal/pgbackrest/pgbackrest_test.go (hunk 1 above)
Comment (tony-landreth) [reply internal/pgbackrest/pgbackrest_test.go:45]: I put the comment on the source, rather than the test.

**Pull request merged**